pkg/cmd/roachtest/test_runner.go

```diff
@@ -1287,6 +1289,26 @@ func (r *testRunner) runTest(
 		s.Run(runCtx, t, c)
 	}()
 
+	// Monitor the task manager for completed events, or failure events and log
+	// them. A failure will call t.Errorf which cancels the test's context.
+	go func() {
```

Comment:
Should we add a mocked roachtest in `test_test.go` to test this logic out? We already have something similar to test timeouts, so I think it should be doable.

Reply:
Good call! I added a test, and it caught a bug! Was using the wrong context for the task manager. Test covers that scenario now as well.